### DISCUSSION TITLE
LAQAの書き方

### DISCUSSION THREAD

**Discussion by nim-hrkn** [Tomoki-YAMASHITA/CrySPY#35] [General]
LAQAの簡単に実行できる例がほしいのです。
soiap_Si8_＊の例はこれでいいのでしょうか。
<pre>
crystal initial.cif ! CIF file for the initial structure
symmetry 1 ! 0: not symmetrize displacements of the atoms or 1: symmetrize

md_mode_cell 3 ! cell-relaxation method
               ! 0: FIRE, 2: quenched MD, or 3: RFC5
number_max_relax_cell 1 ! max. number of the cell relaxation
number_max_relax 100 ! max. number of the atom relaxation
max_displacement 0.1 ! max. displacement of atoms in Bohr

external_stress_v 0.0 0.0 0.0 ! external pressure in GPa

th_force 5d-5 ! convergence threshold for the force in Hartree a.u.
th_stress 5d-7 ! convergence threshold for the stress in Hartree a.u.

force_field 1 ! force field
              ! 1: Stillinger-Weber for Si, 2: Tsuneyuki potential for SiO2,
              ! 3: ZRL for Si-O-N-H, 4: ADP for Nd-Fe-B, 5: Jmatgen, or
              ! 6: Lennard-Jones
</pre>

cryspy.inのnselect_laqa はどういう意味ですか。
[LAQA]
nselect_laqa = 2

**Comment by Tomoki-YAMASHITA** [comment]
LAQAでは最適化ステップごとのデータを取得しているのですが，
soiapではRFC5の時，loopa = 1 の時とそうではない時で出力が異なるので，loopaが1の時のみステップデータを取得するようにしています．
loopaはatomの緩和のループです．
https://github.com/nbsato/soiap/blob/master/doc/instructions.md

上の例ではセルの最適化が1回，atomのrelaxが100回の設定になっているので
LAQAでは1stepデータしか取得されません．

下記のようにすると，毎回セルとatomをrelaxして，そのステップデータを取得できます．
`number_max_relax_cell 20`としていますが
これは選択されたら20ステップだけ最適化を進めるという意味です．
LAQAでは少しずつ最適化します．

```
crystal initial.cif ! CIF file for the initial structure
symmetry 1 ! 0: not symmetrize displacements of the atoms or 1: symmetrize

md_mode_cell 3 ! cell-relaxation method
               ! 0: FIRE, 2: quenched MD, or 3: RFC5
number_max_relax_cell 20 ! max. number of the cell relaxation
number_max_relax 1 ! max. number of the atom relaxation
max_displacement 0.1 ! max. displacement of atoms in Bohr

external_stress_v 0.0 0.0 0.0 ! external pressure in GPa

th_force 5d-5 ! convergence threshold for the force in Hartree a.u.
th_stress 5d-7 ! convergence threshold for the stress in Hartree a.u.

force_field 1 ! force field
              ! 1: Stillinger-Weber for Si, 2: Tsuneyuki potential for SiO2,
              ! 3: ZRL for Si-O-N-H, 4: ADP for Nd-Fe-B, 5: Jmatgen, or
              ! 6: Lennard-Jones
```

**Comment by Tomoki-YAMASHITA** [comment]
nselect_laqa = 2はBOの時のnselect_boと似たようなもので，
一度に選択する構造数を指定します．

tot_struc = 100で始めると，
LAQAではまず100構造全てに対して，1回ジョブをサブミット（上の私が貼った例だと20ステップ）してエネルギーなどのデータを取得します．
その後スコアを計算して，上位n個の構造を選択するのですが，その数が nselect_laqaです．

なお，LAQAではnstageは1にしておかないとエラーが出ます
